go.mod

```diff
@@ -1,9 +1,10 @@
-module go.dagger.io/dagger
+module github.com/dagger/dagger
 
 go 1.18
 
+replace go.dagger.io/dagger => ./sdk/go
```

Comment:
Part of what I need to test, I was **hoping** this would work out of the box without importers having to change anything since it's a `replace` into the same git repository. But I might be day-dreaming?

Reply:
replace rules are not preserved for consumers, and only work locally.

There's a risk there as well; if you use a replace rule, you'll get a "different view of the world" as others.  ideally your tests would run both with replace rule, and without (to verify code works both with code from this repo, and from the actual module that's referenced); see https://github.com/moby/sys/pull/68 (but it's messy).